### PR TITLE
Added http server and made mediaserver return IP when requested through websocket

### PR DESCRIPTION
**Pull request by Speedst1998**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
-#[tokio::main]
-async fn main() {
+#[actix_web::main]
+async fn main() -> std::io::Result<()> {
     env_logger::Builder::from_env(Env::default().default_filter_or("debug")).init();
     info!("Launching app!");
-    // application::run();
     application::start().await;
+    info!("Starting web api");
+    HttpServer::new(|| App::new().service(routes::hello::hello))
```
Comment (ChiD12) [inline src/main.rs]: if we are keeping the application level, this should be there

File: src/service/websocket/socket_manager.rs [hunk 2]
```diff
@@ -81,7 +84,17 @@ impl<'a> SocketManager<'a> {
         }
     }
 
-    async fn blocking_listen(&mut self) -> Result<(), std::io::Error> {
+    async fn get_public_ip(&self) -> String {
+        let result = reqwest::get("http://whatismyip.akamai.com/")
```
Comment (ChiD12) [inline src/service/websocket/socket_manager.rs]: there must be a better way to do this lol, but any external calls should go into a top level folder called clients

**Review by ChiD12 — CHANGES_REQUESTED** [review]
nice job, just a couple things we need to change. API should be on the same file level as service, and it would call a specific service in the service file

File: src/main.rs (hunk 1 above)
Comment (Speedst1998) [reply src/main.rs]: Done!

File: src/service/websocket/socket_manager.rs (hunk 2 above)
Comment (Speedst1998) [reply src/service/websocket/socket_manager.rs]: Hehe it seems cheeky but theres really nothing wrong doing it this way. There are other alternatives though. I think we can query the dns or something, but this was simpler.

Comment (Speedst1998) [reply src/service/websocket/socket_manager.rs]: But I moved it to the clients folder

**Review by ChiD12 — APPROVED** [review]

**Pull request merged**